app/controllers/leads_controller.rb

```diff
@@ -230,15 +252,25 @@ def update_meta
     end
   end
 
+  # PATCH /leads/1/swap.json
+  def swap
+    swap_result = @lead.swap(params['direction'])
+    if swap_result[:leads].empty?
+      @lead.errors.add(:swap, 'failed, attempted to move lead off the end')
+      render json: @lead.errors, status: :unprocessable_entity
+      return
+    end
+
+    @leads = swap_result[:leads]
+    @positions = swap_result[:positions]
+    @futures = @leads.map(&:future)
+  end
+
   # GET /leads/1/otus.json
   def otus
     leads_list = @lead.self_and_descendants.where.not(otu_id: nil).includes(:otu)
```

Comment:
Can you invert this to somethign like `@otus = Otu.joins(:leads).<in key scope>`? We'll need a dynamic scope like that for other uses I suspect.